src/Rector/AbstractRector.php

```diff
@@ -89,11 +89,23 @@ abstract class AbstractRector extends NodeVisitorAbstract implements RectorInter
      */
     private array $deprecatedDependencies = [];
 
+    /**
+     * @var array<class-string, array<string, bool>>
+     */
+    private array $cachedDeprecatedDependenciesWarning = [];
+
     /**
      * Handle deprecated dependencies compatbility
      */
     public function __get(string $name): mixed
     {
+        if (! isset($this->cachedDeprecatedDependenciesWarning[static::class][$name])) {
+            echo sprintf('Get %s property from AbstractRector on %s is deprecated, inject via __construct() instead', $name, static::class);
```

Comment:
I guess this will crash json format?

Reply:
It seems yes, it seems the real solution is remove `__get()` completely imo, just need to have next minor release: 0.19.0 /cc @TomasVotruba

Reply:
maybe you can emit it like a system error?

Reply:
I am proposing to remove `__get()` completely, see https://github.com/rectorphp/rector-src/pull/5070

Reply:
The json format would hide these errors.
This error should be redactored as soon as spotted.